libs/pmp-web/shared/ui-table/src/lib/components/table/table.component.html

```diff
@@ -1,72 +1,36 @@
-<div *ngIf="tableConfig">
-  <mat-table [dataSource]="tableConfig?.data">
+<ng-container *ngIf="_tableConfig">
+  <mat-table [dataSource]="_tableConfig.data">
     <!-- Avatar Column -->
     <ng-container matColumnDef="avatar">
-      <mat-header-cell *matHeaderCellDef class="avatar-column"> </mat-header-cell>
+      <mat-header-cell *matHeaderCellDef class="avatar-column"></mat-header-cell>
       <mat-cell *matCellDef="let element" class="avatar-column">
         <div class="avatar-column__empty-space"></div>
       </mat-cell>
     </ng-container>
 
-    <!-- User Column -->
-    <ng-container matColumnDef="user">
+    <!-- Dynamic Columns -->
+    <ng-container
+      *ngFor="let columnConfig of _tableConfig.columns"
+      [matColumnDef]="columnConfig.name"
+    >
       <mat-header-cell *matHeaderCellDef>
-        <span>{{ 'User' }}</span>
+        <span>{{ columnConfig.label }}</span>
+        <mat-icon *ngIf="columnConfig.isOrderColumn">arrow_upward</mat-icon>
```

Comment:
I think it should be `mat-sort` directive

Reply:
If it will be paginated we can't do anything more than display the information about which column is the source of the sort.